### ISSUE TITLE
Add a new page for current state of ported and non-ported CA rules in FxCop analyzers when compared to CA rules in binary FxCop

### ISSUE THREAD

**Issue by mavasani** [MicrosoftDocs/visualstudio-docs#3192]
https://developercommunity.visualstudio.com/content/problem/517712/ca0507-run-code-analysis-has-been-deprecated-but-m.html: This VS feedback item raises some very important documentation requests for someone who has just become aware about possibility of migration and wants to know what rules have been ported to FxCop analyzers from the original FxCop implementation, what rules have not been ported and if so why, including details on whether the rule has been cut or just not yet implemented, and justification and user feedback(s) on each rule.

I think we need to create a new page for current state of CA rules port in FxCop analyzers, which is linked directly from the top level [landing page](https://docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-analyzers?view=vs-2019) for migration from FxCop to FxCop analyzers:

```
Landing page for Migration from FxCop to FxCop analyzers
 --> Current State of CA rules port in FxCop analyzers
        --> Ported rules
        --> Non-ported rules
              --> Deprecated/Won't port rules
              --> Not yet ported rules (most of these should be lower priority rules, which may get ported based on customer feedback and/or contributions to the repo)
```

1. Ported rules: We already have an auto-generated page that list the rules implemented in FxCop analyzers package: https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.FxCopAnalyzers/Microsoft.CodeAnalysis.FxCopAnalyzers.md. We should either have a pointer to this page OR move all the data from there into this page

2. Non-ported rules: We have created issues in roslyn-analyzers repo tracking porting of each CA rule. You can find them by using this query: https://github.com/dotnet/roslyn-analyzers/issues?q=is%3Aopen+is%3Aissue+label%3AFxCop-Port (we have a special label "FxCop-Port" for these issues) and then search the rule ID in this query - all such issues titles are prefixed with "Port FxCop Rule ...". There is additional documentation at https://github.com/dotnet/roslyn-analyzers/blob/master/docs/FxCopPort/RulesInventory.csv on the triage done prior to the port exercise was started, note that some of the data here is likely stale but still might be a good reference point.
We should potentially list every single non-ported CA rule on this page, and have a pointer to the issue detailing the justification why it has not been ported yet. Again, we can also consider just moving this information into this documentation page if we don't like having pointers to specific github issues from this page.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d8b46f1e-f497-6606-18b5-6e2a4c6d64db
* Version Independent ID: c1e762c4-ae37-541d-f3b9-038cb7041305
* Content: [FxCop code analysis and FxCop analyzers - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-analyzers-faq?view=vs-2019)
* Content Source: [docs/code-quality/fxcop-analyzers-faq.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/fxcop-analyzers-faq.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

**Comment by mavasani** [comment]
Also tagging @mikadumont and @jinujoseph

**Comment by gewarren** [comment]
@mavasani How does this look? I'm just waiting on the info for CA2222 (ported or not), otherwise I think it's pretty much complete.

https://review.docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-rule-port-status?view=vs-2017&branch=pr-en-us-4659

**Comment by mavasani** [comment]
Thanks @gewarren - the page looks really nice and informative. For the unported rules, is there anyway we can also link them to corresponding github issues on the analyzers repo, or maybe somehow convey that users can search for the tracking issue on the analyzers repo? Otherwise, we might end up getting lot of duplicate issues here. However, that section can also come later - the page looks good to me as you have it.

**Comment by gewarren** [comment]
@mavasani This page is live now: https://docs.microsoft.com/en-us/visualstudio/code-quality/fxcop-rule-port-status?view=vs-2019

**Comment by mavasani** [comment]
Thank you @gewarren. That looks very nice!
